lib/dor/text_extraction/cocina_updater.rb

```diff
@@ -0,0 +1,222 @@
+# frozen_string_literal: true
+
+module Dor
+  module TextExtraction
+    # Update Cocina structural metadata with Abbyy output files that have been added to a workspace directory.
+    #
+    # The logic for updating the Cocina is:
+    #
+    # 1. Look at each file in the workspace's content directory.
+    # 2. If an OCR file will overwrite an existing file in Cocina ensure that the file to be
+    #    overwritten was generated by SDR and has not been corrected.
+    # 2. For each file get the file name "stem": e.g. for "abc123.xml" the stem is "abc123".
+    # 3. Look in the Cocina structural metadata for a Resource containing a file with the
+    #    same stem, e.g. "abc123.tiff".
+    # 4. If a match is found add the new file to the Resource.
+    # 5. If a match wasn't found add it to a new Resource.
+    #
+    # rubocop:disable Style/ClassLength
+    class CocinaUpdater
+      # @param dro [Cocina::Models::Dro] the object metadata to update in place
+      # @param workspace_dir [String] the path where the new files to add are
+      def self.update(dro:, workspace_dir:)
+        CocinaUpdater.new(dro:, workspace_dir:).update
+      end
+
+      def initialize(dro:, workspace_dir:)
+        @dro = dro
+        @new_resources = []
+        @workspace_dir = Pathname.new(workspace_dir)
+      end
```

Comment:
I think we can ditch all of the @workspace_dir passing around since we can get it from DruidTools directly in this class.

Reply:
I did try doing that, but the current interface for CocinaUpdater is:

```
CocinaUpdater.update(dro:, workspace_dir:)
```

So it actually was a bit harder to use DruidTools, and not really worth the effort?

Reply:
I'm taking a quick stab at it right now.  We are the only consumer of the class so we can change that signature too.